### PR TITLE
PYTHON-4391: [Bugfix] Use Docker to run LocalAtlas

### PR DESCRIPTION
**Pull request by Jibola**
## Summary
Our former method of using local atlas. (by downloading the binary and then making the $atlas calls) has been failing. We're now moving our operations to leverage podman. As such, we no longer get the same exposed atlas commands and have moved the index creation work to the python `scaffold_atlas.py` script. 

Moreover, I've updated the README to specify a new way to get the URI and pass it as a flag to each script. 

## Expected Errors
Since local atlas had been failing for some time, the errors had hidden other regressions that have popped up in our integrations. As a result, we've examined each and have provided new bug tickets for their resolution.

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ functions:
         working_dir: "src/${DIR}/${REPO_NAME}"
         binary: bash
         env:
-          atlas: ${workdir}/src/atlas/bin/atlas
+          atlas: ${workdir}/src/atlas/bin/atlas
```
Comment (caseyclements) [inline .evergreen/config.yml:58]: we need a linter. :)

File: .evergreen/podman-local-dev.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
```
Comment (caseyclements) [inline .evergreen/podman-local-dev.sh:1]: This appears to be a copy of utils.sh. Am I right?

File: .evergreen/provision-atlas.sh [hunk 3]
```diff
@@ -11,18 +11,11 @@ PING_ATLAS=$EVERGREEN_PATH/ping_atlas.py
 SCAFFOLD_SCRIPT=$EVERGREEN_PATH/scaffold_atlas.py
```
Comment (caseyclements) [inline .evergreen/provision-atlas.sh:11]: There's still a reference to a `ping_atlas.py` here.

**Review by caseyclements — CHANGES_REQUESTED** [review]

File: .evergreen/podman-local-dev.sh (hunk 2 above)
Comment (Jibola) [reply .evergreen/podman-local-dev.sh:1]: Yep. Let me go ahead and... delete this ol' thing.

File: .evergreen/provision-atlas.sh (hunk 3 above)
Comment (Jibola) [inline .evergreen/provision-atlas.sh:11]:
```suggestion
```

File: .evergreen/config.yml (hunk 1 above)
Comment (Jibola) [reply .evergreen/config.yml:58]: Agreed. I'll make a task for it. [PYTHON-4687](https://jira.mongodb.org/browse/PYTHON-4687)

**Review by caseyclements — APPROVED** [review]
Approving this so that we can dig into failures across projects individually.

**Pull request merged**